src/Twig/Loader/LoaderSelector.php

```diff
@@ -34,35 +34,34 @@ public function __construct(LoaderInterface $stringLoader, LoaderInterface $file
         $this->fileLoader = $fileLoader;
     }
 
-    public function getSource(string $name): string
-    {
-        $source = $this->getLoader($name)->getSource($name);
```

Comment:
This method was deprecated on 1.x and removed on 2.0... Not sure what to do with it, and maybe is relevant because of the following lines. (Maybe that's why some things didn't work in sandbox.

Reply:
I would say it's ok to remove it since it's not in the interface anymore.

Reply:
Take a look at my new commit, not sure if this way would be better, If there are no test about how all this should work, it is harded to know if it is broken.

I see two options:
* We either don't remove this method (yet).
* We move this logic to `getSourceContext`.

On both options it would be nice to start introducing some sort of test to see if any of this works.

Reply:
> We either don't remove this method (yet).

What I don't understand, is why not keeping this method would solve any bugs. To me, if the `getSource` method is not in the interface anymore, it's never called by Twig.

> We move this logic to getSourceContext.

The getSource method was introduced 11 years ago. And the getSourceContext 5 years ago https://github.com/sonata-project/SonataFormatterBundle/commit/c045b76ea532372eb55c3c68cac369e5d5c5a513

I don't know when is used the LoaderSelector, but according to https://github.com/sonata-project/SonataFormatterBundle/blob/1684ad4a2459e6e3ba29721fa954bb29a4d4de15/src/DependencyInjection/SonataFormatterExtension.php#L156-L160 it seems like your new fix might be useful.

My only concern would be "why nobody opened an issue in 5 years".

Reply:
lets keep it this way then?